File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.java
class Solution {
    
        class Pair implements Comparable<Pair>
        {
            int plantTime;
            int growTime;
            
            public Pair(int a, int b)
            {
                plantTime = a;
                growTime = b;
            }
            
            
            @Override
            public String toString()
            {
                return "[" + plantTime + " " + growTime + "]";
            }
            
            @Override
            public int compareTo(final Pair pair)
            {
                return pair.growTime - this.growTime;
            }
                
        }
    
    
    public int earliestFullBloom(int[] plantTime, int[] growTime) {
        

        PriorityQueue<Pair> priorityQueue = new PriorityQueue<Pair>();
        
        
        int n = plantTime.length;
        
        for(int i = 0; i<=n-1; i++)
        {
            priorityQueue.offer(new Pair(plantTime[i], growTime[i]));
        }
        
        
        Pair pair = priorityQueue.poll();
        
        int lastPlant = pair.plantTime - 1;
        int lastBloom = lastPlant + pair.growTime + 1;
        
        
        
        while(!priorityQueue.isEmpty())
        {
            Pair p = priorityQueue.poll();
            lastPlant = lastPlant + p.plantTime;
            lastBloom = Math.max(lastBloom, lastPlant + p.growTime + 1);
        }
        
        
        return lastBloom;
        
        
    }
}

File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.cpp
class Solution {
public:
    static bool cmp(pair<int, int> a, pair<int, int> b)
    {
        return a.second > b.second;
    }
    int earliestFullBloom(vector<int>& plantTime, vector<int>& growTime) {
        multiset<pair<int, int> , decltype(&cmp)> s(&cmp);
        
        int i = 1000000000;
        cout<<i<<endl;
        
        
        
        
        int n = plantTime.size();
        for(int i = 0; i<=n-1; i++)
        {    
            s.insert({plantTime[i], growTime[i]});
        }
        
        auto it = s.begin();
        int lastPlantTime = it->first - 1;
        int lastBlossomTime = lastPlantTime + it->second + 1;
        it++;
        for(; it!=s.end(); it++)
        {
            lastPlantTime = lastPlantTime + it->first;
            lastBlossomTime = max(lastBlossomTime, lastPlantTime + it->second + 1);
        }
        
        return lastBlossomTime;
        
    }
};